include/networkit/matching/BSuitorMatcher.hpp

```diff
@@ -0,0 +1,199 @@
+#ifndef NETWORKIT_MATCHING_B_SUITOR_MATCHER_HPP_
+#define NETWORKIT_MATCHING_B_SUITOR_MATCHER_HPP_
+
+#include <algorithm>
+#include <networkit/graph/Graph.hpp>
+#include <networkit/matching/BMatcher.hpp>
+
+namespace NetworKit {
+
+struct Node {
+    node id;
+    edgeweight weight;
+
+    Node() = default;
+    Node(node n, edgeweight w) : id(n), weight(w) {}
+
+    bool operator==(const Node &other) const { return id == other.id && weight == other.weight; }
+    bool operator!=(const Node &other) const { return id != other.id || weight != other.weight; }
+};
+
+struct Data {
```

Comment:
I suggest using a more precise name for the `Data` struct.

Reply:
Agree, they are awful. I haven't come up with a better name yet, but I'm implementing the parallel iterative version of the algorithm at the moment which doesn't use the T (Proposed) object anymore. With this, I will only have the Suitors member and Suitor struct.

Reply:
Renamed into `NodeMatchesInfo` in cb0aa0e992.